resolwe/flow/tests/test_access_api.py

```diff
@@ -0,0 +1,101 @@
+# pylint: disable=missing-docstring
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import unittest
+
+from django.contrib.auth.models import AnonymousUser
+from django.test import LiveServerTestCase
+
+from guardian.shortcuts import assign_perm
+
+from resolwe.flow.models import Collection, Process, Storage
+from resolwe.test import TransactionProcessTestCase, check_installed, with_docker_executor, with_resolwe_api_host
+
+
+class AccessAPIFromExecutorProcessTestCase(TransactionProcessTestCase, LiveServerTestCase):
+
+    def create_collection(self):
+        """Create a test collection that will be accessed via a test process.
+
+        Assign :class:`~django.contrib.auth.models.AnonymousUser`
+        ``view_collection`` permission to the created collection.
+
+        :return: created test collection
+        :rtype: Collection
+
+        """
+        collection = Collection.objects.create(
+            name='collection-foo',
+            contributor=self.contributor,
+        )
+        assign_perm("view_collection", AnonymousUser(), collection)
+        return collection
+
+    def create_process(self):
+        """Create a test process that will query the API for collections.
+
+        Get the list of all collections using ``curl`` to query the Live
+        Resolwe API host.
+
+        :return: created test process
+        :rtype: Process
+
+        """
+        process = Process.objects.create(
+            name='Test accessing API from process',
+            requirements={'expression-engine': 'jinja'},
+            contributor=self.contributor,
+            type='data:test:api-access',
+            input_schema=[],
+            output_schema=[
+                {'name': 'collection-list', 'type': 'basic:json:'},
+            ],
+            run={
+                'language': 'bash',
+                # NOTE: Resolwe Runtime Utilities are not available in the
+                # resolwe/test:base Docker image so we need to use plain echo
+                'program': r"""
+echo "{\"collection-list\": $(curl $RESOLWE_API_HOST/collection)}"
+"""
+            }
+        )
+        return process
+
+    def setUp(self):
+        """Custom initilization of :class:`~TransactionProcessTestCase`."""
+        # NOTE: We want to skip TransactionProcessTestCase's setUp() and just
+        # call its parent's setUp()
+        # pylint: disable=bad-super-call
+        super(TransactionProcessTestCase, self).setUp()
+
+        self.collection = self.create_collection()
+        self.process = self.create_process()
+
+        self._prepare_files_and_dirs()
+
+    def check_results(self, data):
```

Comment:
Make sure that this fails with more explanatory message if test fail. 

At the moment it fails with `'Required fields "collection-list" not given.'`. For example add `'required': False` to `collection-list` field an check if field is set and storage object exist at the beginning of the method.